src/commands/apextests/list.ts

```diff
@@ -56,25 +71,31 @@ export default class ApextestsList extends SfCommand<ApextestsListResult> {
       throw new Error('Invalid directory.');
     }
 
-    const files = readDir.filter((file) => file.endsWith('.cls') || file.endsWith('.trigger'));
-    const testMethodsNames: string[] = [];
+    const localFiles = readDir.filter((file) => file.endsWith('.cls') || file.endsWith('.trigger'));
+
+    const handler = (fileName: string): void => {
+      const path = `${directory}/${fileName}`;
+      const data = fs.readFileSync(path, 'utf-8');
+      const testMethods = data.match(TEST_NAME_REGEX);
 
-    // read each file and check for the test methods at the top
-    files.map((file) => {
-      const data = fs.readFileSync(`${directory}/${file}`, 'utf8');
+      testMethodsNames.push(...(testMethods ? ApextestsList.parseTestsNames(testMethods) : []));
+    };
 
-      // try to find, with a RegEx, the test methods listed at the top of the
-      // file with @Tests or @TestSuites
-      const testMethods = data.match(/(@(Tests|TestSuites)).+/g);
-      // for each entry, parse the names
-      // const testMethodsNames = testMethods ? ApextestsList.parseTestsNames(testMethods.join(',')) : [];
-      testMethodsNames.push(...(testMethods ? ApextestsList.parseTestsNames(testMethods.join(',')) : []));
-    });
+    const processor = queue((f: string, cb: (error?: Error | undefined) => void) => {
+      handler(f);
+      cb();
+    }, this.getConcurrencyThreshold());
```

Comment:
**praises:** nice, the code will be a bit safer for huge projects with thousands of classes